libs/qec/python/bindings/py_code.cpp

```diff
@@ -77,13 +77,25 @@ class PyCodeHandle : public qec::code {
     if (!py::hasattr(registeredCode, "stabilizers"))
       throw std::runtime_error(
           "Invalid Python QEC Code. Must have self.stabilizers = "
-          "qec.Stabilizers(...). Please provide the stabilizers.");
+          "[cudaq.SpinOperator(...)] (qec.Stabilizers(...)). Please provide "
+          "the stabilizers.");
+    if (!py::hasattr(registeredCode, "pauli_observables"))
+      throw std::runtime_error(
+          "Invalid Python QEC Code. Must have self.pauli_observables = "
+          "[cudaq.SpinOperator(...)]. Please provide the observables.");
     if (!py::hasattr(registeredCode, "operation_encodings"))
       throw std::runtime_error(
           "Invalid Python QEC Code. Must have self.operation_encodings = "
           "{...}. Please provide the CUDA-Q kernels for the operation "
           "encodings.");
 
+    if (py::hasattr(registeredCode, "pauli_observables")) {
+      auto obs_terms = registeredCode.attr("pauli_observables")
+                           .cast<std::vector<cudaq::spin_op_term>>();
+      m_pauli_observables.reserve(obs_terms.size());
+      for (auto &term : obs_terms)
```

Comment:
I suspect you need an rvalue reference here to make the move below work as intended. (I could be wrong, though?)
```suggestion
      for (auto &&term : obs_terms)
```